plugins/org.eclipse.osee.framework.skynet.core/src/org/eclipse/osee/framework/skynet/core/User.java

```diff
@@ -127,18 +133,23 @@ public void toggleFavoriteBranch(BranchId favoriteBranch) {
       saveSettings();
    }
 
+   /**
+    * Cached for efficiency as this is called thousands of times to sort favorites. Cache is updated when User artifact
+    * is changed (eg: transaction in User art is different)
+    */
    public boolean isFavoriteBranch(BranchId branch) {
-      Collection<String> attributes = getAttributesToStringList(CoreAttributeTypes.FavoriteBranch);
-      for (String value : attributes) {
-         try {
-            if (branch.equals(BranchId.valueOf(value))) {
-               return true;
-            }
-         } catch (Exception ex) {
-            // do nothing
+      if (!getTransaction().equals(userModTx)) {
+         if (favoriteBranchIds == null) {
+            favoriteBranchIds = new HashSet<>(20);
```

Comment:
is there a size limit of 20? why is this 20?

Reply:
There to give default size of HashSet if you know it's going to be a certain size so Set doesn't have to be recreated as it grows.  Probably not necessary in this case since most don't have favorites and those that do probably not many.  Removed "20".